packages/sync/build.gradle.kts

```diff
@@ -0,0 +1,303 @@
+    }
+}
+
+kotlin {
+    android("android") {
+        publishLibraryVariants("release", "debug")
+    }
+    sourceSets {
+        getByName("androidMain") {
+            kotlin.srcDir("src/androidMain/kotlin")
+            dependencies {
+                api(project(":library"))
+                api(project(":cinterop"))
+            }
+        }
+
+        getByName("androidTest") {
+            dependencies {
+                implementation(kotlin("test"))
+                implementation(kotlin("test-junit"))
+                implementation("junit:junit:${Versions.junit}")
+                implementation("androidx.test.ext:junit:${Versions.androidxJunit}")
+                implementation("androidx.test:runner:${Versions.androidxTest}")
+                implementation("androidx.test:rules:${Versions.androidxTest}")
+                implementation(kotlin("reflect:${Versions.kotlin}"))
+            }
+        }
+    }
+}
+
+kotlin {
+    sourceSets {
+        create("darwinCommon") {
+            dependsOn(getByName("commonMain"))
+        }
+    }
+}
+
+// IOS Configuration
+kotlin {
+    // For ARM, should be changed to iosArm32 or iosArm64
+    // For Linux, should be changed to e.g. linuxX64
+    // For MacOS, should be changed to e.g. macosX64
+    // For Windows, should be changed to e.g. mingwX64
+    ios()
+    sourceSets {
+        getByName("iosMain") {
+            dependsOn(getByName("darwinCommon"))
+        }
+        getByName("iosX64Main") {
+            dependsOn(getByName("iosMain"))
+        }
+        getByName("iosArm64Main") {
+            dependsOn(getByName("iosMain"))
+        }
+        getByName("iosTest") {
+        }
+    }
+}
+
+// Macos configuration
+kotlin {
+    // For ARM, should be changed to iosArm32 or iosArm64
+    // For Linux, should be changed to e.g. linuxX64
+    // For MacOS, should be changed to e.g. macosX64
+    // For Windows, should be changed to e.g. mingwX64
+    macosX64("macos") {}
+    sourceSets {
+        getByName("macosMain") {
+            dependsOn(getByName("darwinCommon"))
+        }
+        getByName("macosTest") {
+        }
+    }
+}
+
+// Needs running emulator
+// tasks.named("iosTest") {
+//    val device: String = project.findProperty("iosDevice")?.toString() ?: "iPhone 11 Pro Max"
+//    dependsOn(kotlin.targets.getByName<KotlinNativeTargetWithSimulatorTests>("ios").binaries.getTest("DEBUG").linkTaskName)
+//    group = JavaBasePlugin.VERIFICATION_GROUP
+//    description = "Runs tests for target 'ios' on an iOS simulator"
+//
+//    doLast {
+//        val binary = kotlin.targets.getByName<KotlinNativeTargetWithSimulatorTests>("ios").binaries.getTest("DEBUG").outputFile
+//        exec {
+//            commandLine("xcrun", "simctl", "spawn", device, binary.absolutePath)
+//        }
+//    }
+// }
+
+realmPublish {
+    pom {
+        name = "Sync"
+        description = "Sync code for Realm Kotlin. This artifact is not " +
```

Comment:
I guess this is not true at the moment. We have a `TODO AUTO-SETUP` for these things that needs to be aligned when we inject the dependencies.